systems/ros/ros_subscriber_system.h

```diff
@@ -0,0 +1,235 @@
+#pragma once
+
+#include <condition_variable>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "drake/common/drake_copyable.h"
+#include "drake/systems/framework/leaf_system.h"
+
+#include "ros/ros.h"
+
+namespace dairlib {
+namespace systems {
+
+/**
+ * Receives ROS messages from a given topic and outputs them to a
+ * System<double>'s port. This class stores the most recently processed ROS
+ * message in the State. When a ROS message arrives asynchronously, an update
+ * event is scheduled to process the message and store it in the State at the
+ * earliest possible simulation time. The output is always consistent with the
+ * State.
+ *
+ * To process a ROS message, CalcNextUpdateTime() needs to be called first to
+ * check for new messages and schedule a callback event if a new ROS message
+ * has arrived. The message is then processed and stored in the Context by
+ * CalcDiscreteVariableUpdates() or CalcUnrestrictedUpdate() depending on the
+ * output type. When this system is evaluated by the Simulator, all these
+ * operations are taken care of by the Simulator. On the other hand, the user
+ * needs to manually replicate this process without the Simulator.
+ *
+ * (Direct clone of LcmSubscriberSystem with pared-down features.)
+ */
+template <typename RosMessage>
+class RosSubscriberSystem : public drake::systems::LeafSystem<double> {
+ public:
+  DRAKE_NO_COPY_NO_MOVE_NO_ASSIGN(RosSubscriberSystem)
+
+  /**
+   * A factory method that returns an %RosSubscriberSystem that takes
+   * Value<RosMessage> message objects on its sole abstract-valued input port.
+   *
+   * @tparam RosMessage message type to serialize.
+   *
+   * @param[in] topic The ROS topic to subscribe to.
+   *
+   * @param node_handle ROS node handle for context (to make the subscriber).
+   */
+  static std::unique_ptr<RosSubscriberSystem<RosMessage>> Make(
+      const std::string& topic, ros::NodeHandle* node_handle) {
+    return std::make_unique<RosSubscriberSystem<RosMessage>>(topic,
+                                                             node_handle);
+  }
+
+  /**
+   * Constructor that returns a subscriber System that provides message objects
+   * on its sole abstract-valued output port.  The type of the message object is
+   * determined by the @p serializer.
+   *
+   * @param[in] channel The ROS topic on which to subscribe.
+   *
+   * @param node_handle The ROS context.
+   */
+  RosSubscriberSystem(const std::string& topic, ros::NodeHandle* node_handle)
+      : topic_(topic), node_handle_(node_handle) {
+    DRAKE_DEMAND(node_handle_);
+
+    subscriber_ = node_handle->subscribe(
+        topic, 100, &RosSubscriberSystem<RosMessage>::HandleMessage, this);
```

Comment:
Similar question to publisher system on static queue size

Reply:
This queue should probably be of size 1, as it is with LCM.